ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/translator/TypeConverter.java

```diff
@@ -160,7 +160,7 @@ public static RelDataType convert(TypeInfo type, RelDataTypeFactory dtFactory)
       convertedType = convert((UnionTypeInfo) type, dtFactory);
       break;
     }
-    return convertedType;
+    return dtFactory.createTypeWithNullability(convertedType, true);
```

Comment:
This makes sense because Hive does not have a concept of not nullable type. Can we add a comment?